keps/sig-storage/177-volume-snapshot/README.md

```diff
@@ -231,24 +251,305 @@ Here are the changes since the original design proposal:
+    heartbeats, leader election, etc.)
+    leader election
+
+* **Will enabling / using this feature result in introducing new API types?**
+  Describe them, providing:
+  - API type: volumesnapshots, volumesnapshotcontents, and volumesnapshotclasses are CRDs
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+  If a cloud provider has implemented CSI snapshot functionality, it will be called when snapshot operations are triggered.
+
+* **Will enabling / using this feature result in increasing size or count of
+the existing API objects?**
+  Describe them, providing:
+  - API type(s): New CRDs volumesnapshots, volumesnapshotcontents, and volumesnapshotclasses are added
+  - Estimated increase in size: (e.g., new annotation of size 32B)
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+    Each PV can have multiple snapshots. The limit depends on individual storage systems.
+
+* **Will enabling / using this feature result in increasing time taken by any
+operations covered by [existing SLIs/SLOs]?**
+  Think about adding additional work or introducing new steps in between
+  (e.g. need to do X to start a container), etc. Please describe the details.
+  This is a new feature so it will not increase time taken by existing operations.
+
+* **Will enabling / using this feature result in non-negligible increase of
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  Things to keep in mind include: additional in-memory state, additional
+  non-trivial computations, excessive access to disks (including increased log
+  volume), significant amount of data sent and/or received over network, etc.
+  This through this both in small and large cases, again with respect to the
+  [supported limits].
+  This will add new snapshot API objects to the API server. IO depends on the underlying storage system. We only provide a snapshot API at control layer. Copy on write is typically used by snapshot technologies, but our snapshot API does not make any requirements on that because it depends on individual storage systems.
+
+### Troubleshooting
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+  If API server is unavailable, an API update will fail due to timeout. Failed operations will be added back to a rate limited queue for retries.
+  A CreateSnapshot call to CSI driver is idemponent. So when API server is back and the same request is sent to the CSI driver again, the CSI driver should return the results from the same snapshot.
```

Comment:
typo: idempotent

Reply:
Good catch.  I wonder why it is not caught by the spell check which should be part of the verify script.